internal/pkg/vm/internal/vmcontext/vmcontext.go

```diff
@@ -313,6 +298,10 @@ func send(ctx context.Context, transfer TransferFn, vmCtx ExtendedRuntime) ([][]
 		panic("trying to execute fake method on the actual VM, fix test")
 	}
 
+	if msg.Method == types.ConstructorMethodID && !vmCtx.From().Code.Equals(types.InitActorCodeCid) {
```

Comment:
Without this, a network user could call an actor's constructor directly and reinitialize it's state. Conceivably the constructor method itself could make this check, but constructors should be pretty consistent so this is safer.